packages/session-replay-browser/src/session-replay.ts

```diff
@@ -171,26 +171,37 @@ export class SessionReplay implements AmplitudeSessionReplay {
     void this.initialize();
   };
 
-  evaluateTargetingAndRecord = async () => {
+  evaluateTargetingAndRecord = async (options?: { event?: Event }) => {
     if (!this.identifiers || !this.identifiers.sessionId || !this.config) {
       this.loggerProvider.error('Session replay init has not been called, cannot evaluate targeting.');
-      return;
+      return false;
+    }
+    let eventForTargeting = options?.event;
+
+    if (
+      [SpecialEventType.GROUP_IDENTIFY, SpecialEventType.IDENTIFY, SpecialEventType.REVENUE].includes(
+        eventForTargeting?.event_type as SpecialEventType,
+      )
+    ) {
+      eventForTargeting = undefined;
     }
 
     let userProperties;
     if (this.config.instanceName) {
       const identityStore = getAnalyticsConnector(this.config.instanceName).identityStore;
       userProperties = identityStore.getIdentity().userProperties;
     }
-    const sessionTargetingMatch = await evaluateTargetingAndStore({
+    this.sessionTargetingMatch = await evaluateTargetingAndStore({
       sessionId: this.identifiers.sessionId,
       config: this.config,
-      targetingParams: { userProperties },
+      targetingParams: { userProperties, event: eventForTargeting },
     });
 
-    if (sessionTargetingMatch) {
+    if (this.sessionTargetingMatch) {
       this.recordEvents();
     }
```

Comment:
Is `recordEvents()` idempotent? Could this function potentially be called more than once causing record events to be triggered multiple times?

Reply:
It can be called more than once, the first thing this function does is call rrweb's cancel recording callback: https://github.com/amplitude/Amplitude-TypeScript/blob/v1.x/packages/session-replay-browser/src/session-replay.ts#L320C19-L320C19. So there can never be more than one recording/MutationObserver observing at once

Reply:
I wonder if that leads to any janky behavior though? That is if we're constantly restarting recording.